internal/workflow/ProjectAPIKeyOperator.java

```diff
@@ -30,41 +36,42 @@ public UUID createProjectAPIKey(UUID projectId, String description) throws Excep
         return newApiKeyId;
     }
 
-    public UUID getOldKeyUUID(UUID projectId, String oldKey) throws Exception {
+    public SimpleEntry<AuthToken, UUID> getProjectAuthTokenByToken(String token) throws Exception {
+        ProjectList projectList = projectsApi.findProjects(null, null, null, null);
+        if (projectList.getProjects().size() != 1) {
+            throw new Exception("Expected only one project to be associated with the token");
+        }
+        UUID projectId = projectList.getProjects().get(0).getId();
+
         AuthTokenList tokenList = authApi.findProjectAPIKeys(projectId, null, null);
-        for (AuthToken token : tokenList.getApiKeys()) {
-            if (token.getToken().equals(oldKey)) {
-                return token.getId();
+        for (AuthToken authToken : tokenList.getApiKeys()) {
+            if (authToken.getToken().equals(token)) {
+                return new SimpleEntry<>(authToken, projectId);
             }
         }
-        throw new Exception("Old API key not found");
+        throw new Exception("Provided token not found");
     }
 
     public void deleteAPIKey(UUID apiKeyId) throws Exception {
         authApi.deleteAPIKey(apiKeyId);
     }
 
-    public static void main(String[] args) {
-        String oldApiKey = "";
-        if (oldApiKey.isEmpty()) {
-            throw new IllegalArgumentException("API Key is not set. Please set the oldApiKey variable.");
-        }
+    public String rotateProjectKey(String authToken) {
+        try {
+            SimpleEntry<AuthToken, UUID> result = getProjectAuthTokenByToken(authToken);
+            AuthToken oldAuthToken = result.getKey();
+            UUID projectId = result.getValue();
 
-        String projectIdString = "";
-        if (projectIdString.isEmpty()) {
-            throw new IllegalArgumentException("Project ID is not set. Please set the projectID variable.");
-        }
+            if (oldAuthToken == null || oldAuthToken.getProject() == null) {
+                throw new Exception("Cannot rotate API key. The current API key is not associated with a project.");
+            }
 
-        ProjectAPIKeyOperator projAPIKeyOperator = new ProjectAPIKeyOperator(oldApiKey);
-        UUID projectId = UUID.fromString(projectIdString);
-        try {
-            projAPIKeyOperator.createProjectAPIKey(projectId, "test-api-key");
+            UUID newApiKeyId = createProjectAPIKey(projectId, "rotated-api-key");
+            deleteAPIKey(oldAuthToken.getId());
```

Comment:
It looks like this is using the old auth token to delete itself.  As a safety measure, can we use the new auth token to delete the old one?  That avoids (or at least reduces the risk of) deleting the old token only to find out later that the new one is somehow invalid.

Reply:
Fixed it. I'm passing the newToken to delete the old one.